src/Analyzers/Core/Analyzers/UseAutoProperty/AbstractUseAutoPropertyAnalyzer.cs

```diff
@@ -43,274 +53,186 @@ protected AbstractUseAutoPropertyAnalyzer()
         protected abstract SyntaxNode GetFieldNode(TFieldDeclaration fieldDeclaration, TVariableDeclarator variableDeclarator);
 
         protected abstract void RegisterIneligibleFieldsAction(
-            List<AnalysisResult> analysisResults, HashSet<IFieldSymbol> ineligibleFields,
-            Compilation compilation, CancellationToken cancellationToken);
+            ConcurrentSet<IFieldSymbol> ineligibleFields, SemanticModel semanticModel, SyntaxNode codeBlock, CancellationToken cancellationToken);
 
         protected sealed override void InitializeWorker(AnalysisContext context)
-            => context.RegisterSemanticModelAction(AnalyzeSemanticModel);
-
-        private void AnalyzeSemanticModel(SemanticModelAnalysisContext context)
-        {
-            var cancellationToken = context.CancellationToken;
-            var semanticModel = context.SemanticModel;
-
-            // Don't even bother doing the analysis if the user doesn't even want auto-props.
-            var option = context.GetAnalyzerOptions().PreferAutoProperties;
-            if (!option.Value)
+            => context.RegisterSymbolStartAction(context =>
             {
-                return;
-            }
+                var namedType = (INamedTypeSymbol)context.Symbol;
+                if (namedType.TypeKind is not TypeKind.Class and not TypeKind.Struct and not TypeKind.Module)
+                    return;
 
-            var analysisResults = new List<AnalysisResult>();
-            var ineligibleFields = new HashSet<IFieldSymbol>();
+                var analysisResults = new ConcurrentQueue<AnalysisResult>();
```

Comment:
>ConcurrentQueue [](http://example.com/codeflow?start=42&length=15)

I'm curious why this was changed to concurrent. Was this for safety safe, or can the registered methods get called back on different threads?

Reply:
> Was this for safety safe, or can the registered methods get called back on different threads?

Yes.  It's for safety because all the callbacks can operate concurrently.